### PR TITLE
Add api docs to pipeline stage deploy end point

### PR DESCRIPTION
**Pull request by knolleary**
Adds api docs to the pipeline stage deploy end point

### CODE REVIEW

File: forge/ee/routes/pipeline/index.js [hunk 1]
```diff
@@ -438,7 +438,35 @@ module.exports = async function (app) {
...
+                type: 'object',
+                properties: {
+                    pipelineId: { type: 'string' },
+                    stageId: { type: 'string' }
+                }
+            },
+            body: {
+                type: 'object',
+                properties: {
+                    sourceSnapshotId: { type: 'string', description: 'The snapshot to deploy if the stage action is set to "prompt"' }
```
Comment (Pezmc) [inline forge/ee/routes/pipeline/index.js:455]: This is optional, does it need to be flagged as such?

Comment (knolleary) [reply forge/ee/routes/pipeline/index.js:455]: Properties are optional by default. Otherwise, we would set the `required` flag on them

**Comment by Pezmc** [comment]
This will need to be rebased to see the test fixes. Will merge after #2818.

**Review by Pezmc — APPROVED** [review]

**Comment by knolleary** [comment]
Ah yes - running with `-b` locally means it only tells me about the first failure 🤦🏻 ... that mystery solved. Still investigating.

**Comment by knolleary** [comment]
Tests should be fixed. Whilst the schema defaulted to `body.sourceSnapshotId` being optional, it still expected a request body to be present, even if empty.

**Comment by knolleary** [comment]
Setting `body.required = false` seems to really upset fastify and causes issues in completely unrelated routes.

Have changed the approach to specifying the type of body as `['object', 'null']` which appears to keep everyone happy.

**Comment by knolleary** [comment]
Tests related to this PR all passing... failures related to issues on main persist.

Once they are resolved, will rebase and force push

**Pull request merged**